src/platforms/macos/daemon/xpcdaemonserver.mm

```diff
@@ -0,0 +1,278 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "xpcdaemonserver.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include "leakdetector.h"
+#include "logger.h"
+#include "platforms/macos/macosutils.h"
+#include "platforms/macos/xpcdaemonprotocol.h"
+#include "version.h"
+
+constexpr const int XPC_SESSION_MAX_BACKLOG = 32;
+
+namespace {
+Logger logger("XpcDaemonServer");
+}  // namespace
+
+// This property exists, but it's private. Make it available:
+@interface NSXPCConnection(PrivateAuditToken)
+@property (nonatomic, readonly) audit_token_t auditToken;
+@end
+
+@interface XpcDaemonDelegate : NSObject<NSXPCListenerDelegate>
+@property Daemon* daemon;
+@property (copy) NSString* teamIdentifier;
+
+- (id)initWithObject:(Daemon*)daemon;
++ (NSString*)getTeamIdentifier:(SecTaskRef)task;
+@end
+
+@interface XpcSessionDelegate : NSObject<XpcDaemonProtocol>
+@property Daemon* daemon;
+@property XpcDaemonSession* bridge;
+
+- (id)initWithObject:(Daemon*)daemon;
+@end
+
+XpcDaemonServer::XpcDaemonServer(Daemon* daemon) : QObject(daemon) {
+  MZ_COUNT_CTOR(XpcDaemonServer);
+
+  QString daemonId = MacOSUtils::appId() + ".daemon";
+  logger.debug() << "XpcDaemonServer created:" << daemonId;
+
+  XpcDaemonDelegate* delegate =
+      [[XpcDaemonDelegate alloc] initWithObject:daemon];
+  NSXPCListener* listener =
+      [[NSXPCListener alloc] initWithMachServiceName:daemonId.toNSString()];
+  listener.delegate = delegate;
+
+  // Connections to the daemon require codesigning
+  // TODO: It would be nice if we could turn this off for developers somehow.
+  if (@available(macOS 13, *)) {
+    constexpr const char* oidExtAppleCodesign =
+        "(certificate leaf[field.1.2.840.113635.100.6.1.13] or" \
+        " certificate leaf[field.1.2.840.113635.100.6.1.12])";
+    constexpr const char* oidExtAppleCaIntermediate =
+        "(certificate 1[field.1.2.840.113635.100.6.2.6] or" \
+        " certificate 1[field.1.2.840.113635.100.6.2.1])";
```

Comment:
Where is this coming from?

Reply:
Thanks for the detailed explanation. Maybe drop a link to this documentation (and one for 6.1.12) in the comments so it's clear to future maintainers where these came from?